actpool/actqueue.go

```diff
@@ -154,13 +155,18 @@ func (q *actQueue) cleanTimeout() []action.SealedEnvelope {
 			removedFromQueue = append(removedFromQueue, q.items[q.index[i].nonce])
 			delete(q.items, q.index[i].nonce)
 			q.index[i] = q.index[size-1]
+			removedIndex = i
 			size--
 			continue
 		}
 		i++
 	}
 	q.index = q.index[:size]
-	heap.Init(&q.index)
+	if len(removedFromQueue) == 1 {
```

Comment:
If the number of items to be removed is larger than 1, the time complexity is still O(n)